event/src/unvalidated/signed/cacao.rs

```diff
@@ -0,0 +1,155 @@
+//! Structures for encoding and decoding CACAO capability objects.
+
+use serde::{Deserialize, Serialize};
+
+/// Capability object, see https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-74.md
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Capability {
+    #[serde(rename = "h")]
+    header: Header,
+    #[serde(rename = "p")]
+    payload: Payload,
+    #[serde(rename = "s")]
+    signature: Signature,
+}
+
+impl Capability {
+    /// Get the header
+    pub fn header(&self) -> &Header {
+        &self.header
+    }
+
+    /// Get the payload
+    pub fn payload(&self) -> &Payload {
+        &self.payload
+    }
+
+    /// Get the signature
+    pub fn signature(&self) -> &Signature {
+        &self.signature
+    }
+}
+/// Header for a CACAO
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Header {
+    #[serde(rename = "t")]
+    r#type: String,
```

Comment:
`r#type`?  Why the weird field name?  Can we also comment what this field is?

Reply:
`type` is a keyword in rust but you can use the `r#` syntax to still use it. I prefer using the full names for things vs abbreviations. As for a comment there isn't much in the spec about its purpose, the getter for the field has a simple comment.